Tests/BasicsTests/TripleTests.swift

```diff
@@ -9,204 +9,244 @@
-            XCTAssert(
-                actualTriple == expectedTriple,
-                """
-                Actual triple '\(actualTriple)' did not match expected triple \
-                '\(expectedTriple)' for platform version '\(version)'.
-                """,
-                file: file,
-                line: line)
+struct TripleTests {
+    @Test(
+        "Triple is Apple and is Darwin",
+        arguments: [
+            (tripleName:"x86_64-pc-linux-gnu", isApple: false, isDarwin: false),
+            (tripleName:"x86_64-pc-linux-musl", isApple: false, isDarwin: false),
+            (tripleName:"powerpc-bgp-linux", isApple: false, isDarwin: false),
+            (tripleName:"arm-none-none-eabi", isApple: false, isDarwin: false),
+            (tripleName:"arm-none-linux-musleabi", isApple: false, isDarwin: false),
+            (tripleName:"wasm32-unknown-wasi", isApple: false, isDarwin: false),
+            (tripleName:"riscv64-unknown-linux", isApple: false, isDarwin: false),
+            (tripleName:"mips-mti-linux-gnu", isApple: false, isDarwin: false),
+            (tripleName:"mipsel-img-linux-gnu", isApple: false, isDarwin: false),
+            (tripleName:"mips64-mti-linux-gnu", isApple: false, isDarwin: false),
+            (tripleName:"mips64el-img-linux-gnu", isApple: false, isDarwin: false),
+            (tripleName:"mips64el-img-linux-gnuabin32", isApple: false, isDarwin: false),
+            (tripleName:"mips64-unknown-linux-gnuabi64", isApple: false, isDarwin: false),
+            (tripleName:"mips64-unknown-linux-gnuabin32", isApple: false, isDarwin: false),
+            (tripleName:"mipsel-unknown-linux-gnu", isApple: false, isDarwin: false),
+            (tripleName:"mips-unknown-linux-gnu", isApple: false, isDarwin: false),
+            (tripleName:"arm-oe-linux-gnueabi", isApple: false, isDarwin: false),
+            (tripleName:"aarch64-oe-linux", isApple: false, isDarwin: false),
+            (tripleName:"armv7em-unknown-none-macho", isApple: false, isDarwin: false),
+            (tripleName:"armv7em-apple-none-macho", isApple: true, isDarwin: false),
+            (tripleName:"armv7em-apple-none", isApple: true, isDarwin: false),
+            (tripleName:"aarch64-apple-macosx", isApple: true, isDarwin: true),
+            (tripleName:"x86_64-apple-macosx", isApple: true, isDarwin: true),
+            (tripleName:"x86_64-apple-macosx10.15", isApple: true, isDarwin: true),
+            (tripleName:"x86_64h-apple-darwin", isApple: true, isDarwin: true),
+            (tripleName:"i686-pc-windows-msvc", isApple: false, isDarwin: false),
+            (tripleName:"i686-pc-windows-gnu", isApple: false, isDarwin: false),
+            (tripleName:"i686-pc-windows-cygnus", isApple: false, isDarwin: false)
+        ]
+    )
+    func isAppleIsDarwin(_ tripleName: String, _ isApple: Bool, _ isDarwin: Bool) {
+        guard let triple = try? Triple(tripleName) else {
```

Comment:
As below (oops) use `try #require` and don't use `try?`.

Reply:
ACK